tools/GenerateDocFxStructure/Program.cs

```diff
@@ -0,0 +1,313 @@
+
+namespace GenerateDocFxStructure
+{
+    class Program
+    {
+        private static readonly string[] _mediaExtentions = new string[] { ".png", ".jpg", ".jpeg", ".mpg", ".mpeg", ".mov", ".mp4", ".bmp", ".gif", ".fzz" };
+        private static CommandlineOptions? _options;
+        private static int _returnvalue;
+        private static MessageHelper? _message;
+
+        private static List<string> allFiles = new List<string>();
+
+        private static int Main(string[] args)
+        {
+            Console.WriteLine("Hello World!");
+            try
+            {
+                Parser.Default.ParseArguments<CommandlineOptions>(args)
+                                   .WithParsed<CommandlineOptions>(RunLogic)
+                                   .WithNotParsed(HandleErrors);
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"ERROR: Parsing arguments threw an exception with message `{ex.Message}`");
+                _returnvalue = 1;
+            }
+
+            Console.WriteLine($"Exit with return code {_returnvalue}");
+
+            return _returnvalue;
+        }
+
+        private static void RunLogic(CommandlineOptions o)
+        {
+            _message = new MessageHelper(o);
+            if (string.IsNullOrEmpty(o.DestinationFolder))
+            {
+                _message.Error($"Destination folder must be specified.");
+                _returnvalue = 1;
+                return;
+            }
+
+            if (string.IsNullOrEmpty(o.SourceFolder))
+            {
+                _message.Error($"Source folder must be specified.");
+                _returnvalue = 1;
+                return;
+            }
+
+            if (string.IsNullOrEmpty(o.MediaFolder))
+            {
+                _message.Error($"Media folder must be specified.");
+                _returnvalue = 1;
+                return;
+            }
+
+            o.DestinationFolder = Path.GetFullPath(o.DestinationFolder);
+            o.MediaFolder = Path.GetFullPath(o.MediaFolder);
+            o.SourceFolder = Path.GetFullPath(o.SourceFolder);
+            _options = o;
+
+            _message.Verbose($"Source folder     : {_options.SourceFolder}");
+            _message.Verbose($"Destination folder: {_options.DestinationFolder}");
+            _message.Verbose($"Media folder      : {_options.MediaFolder}");
+            _message.Verbose($"Verbose           : {_options.Verbose}");
+
+            if (!Directory.Exists(_options.SourceFolder))
+            {
+                _message.Error($"ERROR: Documentation folder '{_options.SourceFolder}' doesn't exist.");
+                _returnvalue = 1;
+                return;
+            }
+
+            // We will start to get all the files
+            foreach (string file in Directory.EnumerateFiles(_options.SourceFolder, "*.*", SearchOption.AllDirectories))
+            {
+                allFiles.Add(file.ToLowerInvariant());
```

Comment:
I'd avoid normalizing file names - I'd rather show warning if the name isn't all lower case or uses unexpected characters